### PR TITLE
Move publish tasks to use plugin

### PR DESCRIPTION
**Pull request by rafaeltoledo**
This unblocks the upgrade to Gradle 7.x / AGP 7.x. Adapted from https://github.com/uber/okbuck/pull/967

### CODE REVIEW

**Comment by TonyTangAndroid** [comment]
@rafaeltoledo Thank you for working on this task.  In case you are running into issue making the ci pass, here is a sample pull request from RIBs that addresses this tasks.
https://github.com/uber/RIBs/pull/477/files
https://github.com/uber/RIBs/pull/478/files

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,18 +6,18 @@ jobs:
   build:
     name: JDK ${{ matrix.java_version }}
     runs-on: ubuntu-latest
-    strategy: 
+    strategy:
       matrix:
-        # TODO Add 9, 10, 11, 12, and 13 after Kotlin 1.3.60
-        java_version: [8]
+        java_version: [11]
```
Comment (TonyTangAndroid) [inline .github/workflows/ci.yml:11]: If possible, could we introduce a minor diff to update this after we resolve the `gradle publish task` like this pull request. https://github.com/uber/RIBs/pull/477/files

Comment (rafaeltoledo) [reply .github/workflows/ci.yml]: Sure, I removed the changes related to AGP / Gradle / Java, and we can do it on another PR once this one is merged

**Review by TonyTangAndroid — APPROVED** [review]
Thank you for working on this. Looks good to me.

**Pull request merged**